GoogleForCreators/web-stories-wp
# Quick Action Menu: Background Media

## Context

Adds the quick menu specific to background images (also video), this is behind a feature flag.

## Summary

New quick menu for background media (applies to images, video, gifs, whatever media is your background element). 
Has the ability to:
- Navigate to media panel (which ever the existing media came from) 
- Navigate to animation panel 
- If there are animations or filters applied to the bg media element then the 'clear' button will reset those values (hitting undo will bring them back)

## Relevant Technical Choices

Most of the core functionality here was already done so I focused on some peripheral updates to make sharing code easier. 

- moved all the static text and constants to `./constants` to make `useQuickActions` a little less long. 
- update function that was formerly `clearAnimations` to be `handleClearAnimationsAndFilters` (though it doesn't need to be limited to that now) and wrote some helpers to figure out what values we want to reset dynamically. Those live in `../quickActions/utils`
- Updated `handleFocusMedia` so that it can dynamically pick whichever media tab the existing media came from, defaults to local media. Updated foreground image to use this. 
- Adjust checks for the empty bg menu (defualt) vs bg element with media for what menu to display.

## Notes to discuss next week when design is back!

- I feel like maybe some of the copy is out of sorts between the `clear ___` buttons and the snackbar messages. I just checked the figma files today and the snackbar messages say a blanket "styles have been removed" but we're only ever getting rid of either animations or filters so that's a bit misleading. I think that'll be easier to talk to design about next week. 
- On a similar note I think maybe only saying "clear animations" or "clear filters" or "clear animations and filters" on the clear tooltip would be helpful according to what the element has set. 

## To Do

- [ ]  Add in some axe karma tests 💪 

## User-facing changes

None, hidden behind a feature flag 

## Testing Instructions

1. Make sure that `Enable Quick Action Menus` experiment is on. 
2. Open a story or make a new one 
3. Add a background image/video/gif and see the quick menu update to have the options of 1) replace background 2) Add animation 3) clear filters and animation

By default `clear` should be disabled until you have filters or animations present

Clicking `replace background` takes you to the media panel from which the existing background media came from (this is new) 
Clicking `add animation` takes you to the animation panel 
Clicking `clear` (when enabled) will clear off any filter or animation applied to the element and show a snackbar saying what it updated (filter or animation or both). 


https://user-images.githubusercontent.com/10720454/119204990-cc6de600-ba4b-11eb-9f67-3f7483050463.mp4


### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6142
